velox/functions/prestosql/InPredicate.cpp

```diff
@@ -13,12 +13,66 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include "velox/expression/DecodedArgs.h"
 #include "velox/expression/VectorFunction.h"
 #include "velox/type/Filter.h"
 
 namespace facebook::velox::functions {
 namespace {
 
+class GenericInPredicate : public exec::VectorFunction {
+ public:
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    auto* value = decodedArgs.at(0);
+    auto* valueBase = value->base();
+
+    auto* inList = decodedArgs.at(1);
+
+    auto* inListBaseArray = inList->base()->as<ArrayVector>();
+    auto& inListElements = inListBaseArray->elements();
+
+    context.ensureWritable(rows, BOOLEAN(), result);
+    result->clearNulls(rows);
+    auto* boolResult = result->asUnchecked<FlatVector<bool>>();
+
+    rows.applyToSelected([&](vector_size_t row) {
+      if (value->isNullAt(row) || inList->isNullAt(row)) {
+        boolResult->setNull(row, true);
+        return;
+      }
+
+      const auto valueBaseRow = value->index(row);
+
+      const auto arrayRow = inList->index(row);
+      const auto offset = inListBaseArray->offsetAt(arrayRow);
+      const auto size = inListBaseArray->sizeAt(arrayRow);
+
+      bool hasNull = false;
+      for (auto i = 0; i < size; ++i) {
+        if (inListElements->isNullAt(offset + i)) {
+          hasNull = true;
```

Comment:
Shall we break here? or return here? And then set boolResult to true at the end?

Reply:
We cannot break here. We need to keep looking in case there is a match later in the list.